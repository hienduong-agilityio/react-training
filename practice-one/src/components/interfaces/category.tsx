/* eslint-disable semi */
/**
 * Represents the category data structure.
 */
export default interface ICategory {
  id: number;
  name: string;
}
