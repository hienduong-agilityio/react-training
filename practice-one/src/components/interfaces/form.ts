export interface IFormValue {
  name: string;
  price: string;
  description: string;
  category: string;
}
