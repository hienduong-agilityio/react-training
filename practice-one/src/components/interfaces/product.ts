/**
 * Represents the product data structure.
 */
export interface IProduct {
  id: number;
  name: string;
  price: string;
  description: string;
}

export interface IProductByCategory extends IProduct {
  categoryName: string;
}

export type IProductWithoutId = Omit<IProductByCategory, 'id'>;
