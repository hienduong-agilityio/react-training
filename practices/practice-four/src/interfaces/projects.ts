import { STATUS } from '@/enums/status';

interface IProjectsQueryResult {
  // data: The fetched project data, an array of project items.
  data?: IProjectItemProps[];
  // isFetching: A boolean indicating whether the query is currently in progress.
  isFetching: boolean;
  // isFetched: A boolean indicating whether the query has been completed.
  isFetched: boolean;
  // error:  The error object if the query fails, or null if there's no error.
  error: Error | null;
}

interface IProjectItemProps {
  // id: The unique identifier for the project.
  id: string;
  // projectName: The name of the project.
  projectName: string;
  // status: The current status of the project or string, indicating progress or risk level.
  status: STATUS | string;
  // managerId: The id for managers objects.
  managerId?: number;
  // lastUpdate: The date when the project was last updated.
  lastUpdate: string;
  // resources: Optional resources associated with the project, displayed as tags.
  resources?: string;
  // timeline: Optional timeline object containing start and end times for the project.
  timeline?: ITimelineProps;
  // budget: Optional estimated cost of the project, displayed as a financial estimate.
  budget?: number;
  // manager: The manager object for project details.
  manager?: IManagerProps;
}

interface IManagerProps {
  // id: The unique identifier for the manager.
  id?: string;
  // managerName: The name of the project manager responsible for overseeing the project.
  managerName: string;
  // managerImage: Optional URL for the project manager's avatar image.
  managerImage?: string;
}

interface ITimelineProps {
  // timeStart: Optional start time for the timeline. Defaults to an empty string if not provided.
  timeStart?: string;
  // timeEnd: Optional end time for the timeline. Defaults to an empty string if not provided.
  timeEnd?: string;
}

export type { IProjectsQueryResult, IProjectItemProps, IManagerProps, ITimelineProps };
