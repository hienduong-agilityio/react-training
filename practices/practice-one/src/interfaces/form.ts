export interface IFormValue {
  name: string;
  price: number;
  description: string;
  category: string;
}

export interface IValidationMessages extends Omit<IFormValue, 'price'> {
  price: string;
}
