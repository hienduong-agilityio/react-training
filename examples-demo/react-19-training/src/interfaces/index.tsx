// Libraries
import { ReactNode } from 'react';

export interface IFormState {
  errors: Record<string, string>;
}

export interface IBaseComponentProps {
  customClass?: Partial<
    Record<
      'container' | 'inputContainer' | 'input' | 'button' | 'overlay' | 'confirmButton' | 'cancelButton' | 'wrapper',
      string
    >
  >;
}

export interface IInputGroupProps extends IBaseComponentProps {
  placeholder?: string;
  value?: string;
  buttonText?: string;
  inputName?: string;
  isDisabled?: boolean;
  isLoading?: boolean;
  startIcon?: ReactNode;
  onSubmit?: (event: React.FormEvent<HTMLFormElement>) => void;
}

export interface IModalProps extends IBaseComponentProps {
  isOpen: boolean;
  onClose: () => void;
}

export interface IFormFieldProps extends Omit<IInputProps, 'onFocus' | 'onBlur' | 'onMouseEnter' | 'onMouseLeave'> {
  id: string;
  name: string;
  handleInputEvent: (field: string, eventType: 'focus' | 'blur' | 'mouseenter' | 'mouseleave', value?: string) => void;
  getContainerClass: (field: string, hasError: boolean) => string;
  handleFieldChange?: (value: string) => void;
}

import { SVGProps } from 'react';

export interface IIconProps extends SVGProps<SVGSVGElement> {
  size?: number;
  itemCount?: number;
  isStarred?: boolean;
  rating?: number;
}

import { STATUS } from '../enums/status';
import { IInputProps } from '../components/InputField';

interface IProjectsQueryResult {
  // data: The fetched project data, an array of project items.
  data?: IProjectItemProps[];
  // isFetching: A boolean indicating whether the query is currently in progress.
  isFetching: boolean;
  // isFetched: A boolean indicating whether the query has been completed.
  isFetched: boolean;
  // error:  The error object if the query fails, or null if there's no error.
  error: Error | null;
}

interface IProjectItemProps {
  // id: The unique identifier for the project.
  id: string;
  // projectName: The name of the project.
  projectName: string;
  // status: The current status of the project or string, indicating progress or risk level.
  status: STATUS | string;
  // managerId: The id for managers objects.
  managerId?: number;
  // lastUpdate: The date when the project was last updated.
  lastUpdate: string;
  // resources: Optional resources associated with the project, displayed as tags.
  resources?: string;
  // timeline: Optional timeline object containing start and end times for the project.
  timeline?: ITimelineProps;
  // budget: Optional estimated cost of the project, displayed as a financial estimate.
  budget?: number;
  // manager: The manager object for project details.
  manager?: IManagerProps;
}

interface IManagerProps {
  // id: The unique identifier for the manager.
  id?: string;
  // managerName: The name of the project manager responsible for overseeing the project.
  managerName: string;
  // managerImage: Optional URL for the project manager's avatar image.
  managerImage?: string;
}

interface ITimelineProps {
  // timeStart: Optional start time for the timeline. Defaults to an empty string if not provided.
  timeStart?: string;
  // timeEnd: Optional end time for the timeline. Defaults to an empty string if not provided.
  timeEnd?: string;
}

export type { IProjectsQueryResult, IProjectItemProps, IManagerProps, ITimelineProps };
