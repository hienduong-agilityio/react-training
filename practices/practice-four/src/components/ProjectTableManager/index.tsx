// Libraries
import { ChangeEvent, memo } from 'react';

// Components
import { ProjectTable, SelectField, Pagination, DeleteProjectModal, Spinner } from '@/components';

// Interfaces
import { IProjectItemProps, IProjectsQueryResult } from '@/interfaces';

export interface IProjectTableManagerProps {
  isPending: boolean;
  allProjectsQuery: IProjectsQueryResult;
  projectData: IProjectItemProps[];
  rowsPerPage: number;
  totalPages: number;
  currentPage: number;
  selectedProjectId: string;
  isDeleteModalOpen: boolean;
  rowPerPageOption: JSX.Element[];
  handleRowsPerPageChange: (event: ChangeEvent<HTMLSelectElement>) => void;
  handlePageChange: (newPage: number) => void;
  handleOpenDeleteProjectModal: (projectId: string) => void;
  handleDeleteModalClose: () => void;
}

const ProjectTableManager = ({
  isPending,
  allProjectsQuery,
  projectData,
  rowsPerPage,
  totalPages,
  currentPage,
  selectedProjectId,
  isDeleteModalOpen,
  rowPerPageOption,
  handleRowsPerPageChange,
  handlePageChange,
  handleOpenDeleteProjectModal,
  handleDeleteModalClose
}: IProjectTableManagerProps) => {
  return (
    <div className='relative'>
      {isPending && (
        <div className='absolute inset-0 z-10 flex justify-center items-center bg-white/20 backdrop-blur-[1px] pointer-events-auto'>
          <Spinner />
        </div>
      )}

      <div className={isPending ? 'pointer-events-none select-none' : ''}>
        <ProjectTable dataTable={projectData} onDeleteProject={handleOpenDeleteProjectModal} />

        <div className='mt-4 flex items-center justify-between px-5'>
          <span className='text-sm text-gray-600'>
            <strong>{projectData.length ? `1 - ${projectData.length}` : '0'}</strong> of{' '}
            <strong>{allProjectsQuery.data?.length || 0}</strong>
          </span>

          <div className='flex items-center gap-3'>
            <label htmlFor='rowsPerPageSelect'>Rows per page:</label>
            <SelectField
              id='rowsPerPageSelect'
              defaultValue={rowsPerPage}
              onChange={handleRowsPerPageChange}
              customClasses='rounded-md'
            >
              {rowPerPageOption}
            </SelectField>
            <Pagination currentPage={currentPage} totalPages={totalPages} onPageChange={handlePageChange} />
            <DeleteProjectModal
              projectId={selectedProjectId}
              isModalOpen={isDeleteModalOpen}
              onCloseModal={handleDeleteModalClose}
            />
          </div>
        </div>
      </div>
    </div>
  );
};

export default memo(ProjectTableManager);
