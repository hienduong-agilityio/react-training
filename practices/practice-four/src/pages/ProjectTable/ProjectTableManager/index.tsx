// Libraries
import { ChangeEvent, RefObject } from 'react';

// Interfaces
import { IProjectItemProps, IProjectsQueryResult } from '@/interfaces';

// Components
import {
  ProjectFilterDropdown,
  SearchBox,
  Button,
  Spinner,
  ProjectTable,
  SelectField,
  Pagination,
  DeleteProjectModal
} from '@/components';

// Enums
import { BUTTON_VARIANTS, BUTTON_SIZES } from '@/enums';

// Constants
import { SEARCH_OPTION } from '@/constants';

export interface IProjectTableManagerProps {
  //searchField: The current search field being used for filtering (e.g., projectName, status)
  searchField: string;
  //searchKeyword: The current search keyword typed by the user
  searchKeyword: string;
  //searchBoxRef: Reference to the search input field
  searchBoxRef: RefObject<HTMLInputElement | null>;
  //isPending: Indicates whether the project data is loading
  isPending: boolean;
  //allProjectsQuery: Contains the result of the project data query
  allProjectsQuery: IProjectsQueryResult;
  //projectData: The list of project data being displayed in the table
  projectData: IProjectItemProps[];
  //rowsPerPage: The number of rows to display per page
  rowsPerPage: number;
  //totalPages: The total number of pages based on the project data
  totalPages: number;
  //currentPage: The current page being viewed
  currentPage: number;
  //selectedProjectId: The ID of the project selected for deletion
  selectedProjectId: string;
  //isDeleteModalOpen: Whether the delete confirmation modal is open
  isDeleteModalOpen: boolean;
  //rowPerPageOption: The row per page options to display in the dropdown
  rowPerPageOption: JSX.Element[];
  //handleNewProjectClick: Function to handle when the "New Project" button is clicked
  handleNewProjectClick: () => void;
  //handleButtonSearch: Function to trigger the search when the search button is clicked
  handleButtonSearch: () => void;
  //handleDropDownChange: Function to handle changes in the project filter dropdown
  handleDropDownChange: (event: string) => void;
  //handleRowsPerPageChange: Function to handle changes in the rows per page dropdown
  handleRowsPerPageChange: (event: ChangeEvent<HTMLSelectElement>) => void;
  //handlePageChange: Function to handle page changes for pagination
  handlePageChange: (newPage: number) => void;
  //handleOpenDeleteProjectModal: Function to open the delete confirmation modal for a specific project
  handleOpenDeleteProjectModal: (projectId: string) => void;
  //handleDeleteModalClose: Function to close the delete confirmation modal
  handleDeleteModalClose: () => void;
}

const ProjectTableManager = ({
  searchField,
  searchKeyword,
  searchBoxRef,
  isPending,
  allProjectsQuery,
  projectData,
  rowsPerPage,
  totalPages,
  currentPage,
  selectedProjectId,
  isDeleteModalOpen,
  rowPerPageOption,
  handleNewProjectClick,
  handleButtonSearch,
  handleDropDownChange,
  handleRowsPerPageChange,
  handlePageChange,
  handleOpenDeleteProjectModal,
  handleDeleteModalClose
}: IProjectTableManagerProps) => {
  return (
    <div>
      <div className='flex justify-between pt-5 pb-7 px-3'>
        <div className='flex h-auto'>
          <ProjectFilterDropdown options={SEARCH_OPTION} onChange={handleDropDownChange} searchField={searchField} />
          <div className='flex gap-2'>
            <SearchBox ref={searchBoxRef} value={searchKeyword} />
            <Button onClick={handleButtonSearch} variant={BUTTON_VARIANTS.OUTLINED} size={BUTTON_SIZES.SMALL}>
              Search
            </Button>
          </div>
        </div>
        <Button
          onClick={handleNewProjectClick}
          className='border-0 px-3 py-1 text-sm bg-primary-500 hover:bg-primary-600 focus:bg-primary-500 text-white flex gap-2 items-center focus:ring-4 focus:ring-primary-100 rounded-md cursor-pointer'
        >
          New Project
        </Button>
      </div>

      {isPending ? (
        <div className='absolute inset-0 flex justify-center items-center'>
          <Spinner />
        </div>
      ) : (
        allProjectsQuery.isFetched && (
          <div>
            <ProjectTable dataTable={projectData || []} onDeleteProject={handleOpenDeleteProjectModal} />
            <div className='mt-4 flex items-center justify-between px-5'>
              <span className='text-sm text-gray-600'>
                <strong>{`${projectData.length > 0 ? 1 : 0} - ${projectData.length}`}</strong> of
                <strong>{` ${allProjectsQuery.data?.length}`}</strong>
              </span>
              <div className='flex items-center justify-end'>
                <label htmlFor='rowsPerPageSelect' className='mr-2'>
                  Rows per page:
                </label>
                <SelectField
                  id='row-per-page-selection'
                  name='rowsPerPageSelect'
                  value={rowsPerPage.toString()}
                  defaultValue={rowsPerPage}
                  onChange={handleRowsPerPageChange}
                  customClasses='rounded-md'
                >
                  {rowPerPageOption}
                </SelectField>
                <Pagination currentPage={currentPage} totalPages={totalPages} onPageChange={handlePageChange} />
                <DeleteProjectModal
                  projectId={selectedProjectId}
                  isModalOpen={isDeleteModalOpen}
                  onCloseModal={handleDeleteModalClose}
                />
              </div>
            </div>
          </div>
        )
      )}
    </div>
  );
};

export default ProjectTableManager;
