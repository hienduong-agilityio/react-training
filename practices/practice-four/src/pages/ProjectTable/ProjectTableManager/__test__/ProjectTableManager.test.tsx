// Libraries
import { render, screen, fireEvent } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

// Pages
import ProjectTableManager, { IProjectTableManagerProps } from '@/pages/ProjectTable/ProjectTableManager';

// Interfaces
import { IProjectsQueryResult } from '@/interfaces';

const defaultProps: IProjectTableManagerProps = {
  searchField: 'projectName',
  searchKeyword: '',
  searchBoxRef: { current: null },
  isPending: false,
  allProjectsQuery: {
    isFetched: true,
    isFetching: false,
    error: null,
    data: []
  } as IProjectsQueryResult,
  projectData: [],
  rowsPerPage: 10,
  totalPages: 1,
  currentPage: 1,
  selectedProjectId: '',
  isDeleteModalOpen: false,
  rowPerPageOption: [
    <option key='10' value='10'>
      10
    </option>,
    <option key='20' value='20'>
      20
    </option>
  ],
  handleNewProjectClick: jest.fn(),
  handleButtonSearch: jest.fn(),
  handleDropDownChange: jest.fn(),
  handleRowsPerPageChange: jest.fn(),
  handlePageChange: jest.fn(),
  handleOpenDeleteProjectModal: jest.fn(),
  handleDeleteModalClose: jest.fn()
};

describe('ProjectTableManager', () => {
  const queryClient = new QueryClient();

  const renderComponent = (props: Partial<IProjectTableManagerProps> = {}) => {
    return render(
      <MemoryRouter>
        <QueryClientProvider client={queryClient}>
          <ProjectTableManager {...defaultProps} {...props} />
        </QueryClientProvider>
      </MemoryRouter>
    );
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders the component and matches snapshot', () => {
    const { asFragment } = renderComponent();
    expect(asFragment()).toMatchSnapshot();
  });

  it('displays the SearchBox, Dropdown, and New Project button', () => {
    renderComponent();

    expect(screen.getByRole('textbox')).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /Search/i })).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /New Project/i })).toBeInTheDocument();

    const searchButton = screen.getByRole('button', { name: /Search/i });
    fireEvent.click(searchButton);

    expect(defaultProps.handleButtonSearch).toHaveBeenCalledTimes(1);

    const newProjectButton = screen.getByRole('button', { name: /New Project/i });
    fireEvent.click(newProjectButton);

    expect(defaultProps.handleNewProjectClick).toHaveBeenCalledTimes(1);
  });

  it('displays a spinner when data is loading', () => {
    renderComponent({ isPending: true });

    expect(screen.getByRole('progressbar')).toBeInTheDocument();
  });

  it('renders the ProjectTable when data is available', () => {
    renderComponent({
      allProjectsQuery: {
        isFetched: true,
        isFetching: false,
        error: null,
        data: [
          {
            id: '1',
            projectName: 'Project 1',
            manager: {
              managerName: 'John Doe'
            }
          }
        ]
      } as IProjectsQueryResult,
      projectData: [
        {
          id: '1',
          projectName: 'Project 1',
          manager: {
            managerName: 'John Doe'
          },
          lastUpdate: '',
          status: '',
          resources: '',
          timeline: {}
        }
      ]
    });

    expect(screen.getByText('Project 1')).toBeInTheDocument();
  });

  it('renders the delete modal when isDeleteModalOpen is true', () => {
    renderComponent({ isDeleteModalOpen: true });
    expect(screen.getByRole('dialog')).toBeInTheDocument();
  });

  it('calls handlePageChange when pagination buttons are clicked', () => {
    renderComponent({
      totalPages: 2,
      currentPage: 1
    });

    const nextPageButton = screen.getByAltText(/Arrow Right/i);
    fireEvent.click(nextPageButton);

    expect(defaultProps.handlePageChange).toHaveBeenCalledTimes(1);
  });
});
