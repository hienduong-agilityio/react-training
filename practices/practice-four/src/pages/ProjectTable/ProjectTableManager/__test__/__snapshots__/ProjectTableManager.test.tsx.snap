// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectTableManager renders the component and matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="flex justify-between pt-5 pb-7 px-3"
    >
      <div
        class="flex h-auto"
      >
        <div
          class="relative inline-block text-left"
        >
          <select
            class="inline-flex items-center py-0 h-[33px] text-sm font-medium text-center border-0 rounded-none rounded-s-lg bg-white hover:border-none focus:ring-4 focus:outline-none focus:ring-primary-100 text-gray-700 shadow"
            role="combobox"
          >
            <option
              value="projectName"
            >
              Project name
            </option>
            <option
              value="status"
            >
              Project status
            </option>
          </select>
        </div>
        <div
          class="flex gap-2"
        >
          <div
            class="relative"
          >
            <img
              alt="Search"
              class="pl-4 p-2 absolute"
              src=""
            />
            <div
              class="flex flex-col"
            >
              <input
                class="border gap-2 block outline-none focus:ring-4 focus:ring-primary-100 rounded-md py-[6px] px-3 text-gray-900 disabled:text-light disabled:bg-white shadow h-full w-full pl-9 rounded-none rounded-e-lg"
                id="searchInput"
                placeholder="Search"
                value=""
              />
            </div>
          </div>
          <button
            class="flex gap-2 items-center focus:ring-4 focus:ring-primary-100 rounded-md cursor-pointer disabled:opacity-45 bg-white border-2 text-black px-3 py-1 text-sm text-black"
          >
            Search
          </button>
        </div>
      </div>
      <button
        class="border-0 px-3 py-1 text-sm bg-primary-500 hover:bg-primary-600 focus:bg-primary-500 text-white flex gap-2 items-center focus:ring-4 focus:ring-primary-100 rounded-md cursor-pointer"
      >
        New Project
      </button>
    </div>
    <div>
      <div
        class="w-full shadow-md sm:rounded-lg"
      >
        <div
          class="overflow-y-auto h-[900px]"
        >
          <table
            class="w-full text-sm text-left rtl:text-right text-gray-500 dark:text-gray-400"
          >
            <thead
              class="bg-gray-50"
            >
              <tr
                class="h-9 uppercase"
              >
                <th
                  class="pl-3 font-medium text-gray-500"
                >
                  Id
                </th>
                <th
                  class="font-medium text-gray-500"
                >
                  Project Name
                </th>
                <th
                  class="text-center font-medium text-gray-500"
                >
                  PM
                </th>
                <th
                  class="font-medium text-gray-500"
                >
                  Status
                </th>
                <th
                  class="font-medium text-gray-500"
                >
                  Last update
                </th>
                <th
                  class="text-center font-medium text-gray-500"
                >
                  Resources
                </th>
                <th
                  class="font-medium text-gray-500"
                >
                  Project timeline
                </th>
                <th
                  class="font-medium text-gray-500"
                >
                  Estimation
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td
                  class="text-center py-4"
                  colspan="8"
                >
                  No results found
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <div
        class="mt-4 flex items-center justify-between px-5"
      >
        <span
          class="text-sm text-gray-600"
        >
          <strong>
            0 - 0
          </strong>
           of
          <strong>
             0
          </strong>
        </span>
        <div
          class="flex items-center justify-end"
        >
          <label
            class="mr-2"
            for="rowsPerPageSelect"
          >
            Rows per page:
          </label>
          <div
            class="flex flex-col"
          >
            <select
              class="border gap-2 block outline-none focus:ring-4 focus:ring-primary-100 rounded-md py-[6px] px-3 text-gray-900 disabled:text-light disabled:bg-white shadow rounded-md"
              id="row-per-page-selection"
              name="rowsPerPageSelect"
            >
              <option
                value="10"
              >
                10
              </option>
              <option
                value="20"
              >
                20
              </option>
            </select>
          </div>
          <div
            class="flex justify-between items-center p-4"
          >
            <div
              class="flex items-center space-x-2"
            >
              <button
                class="px-2 py-2 rounded-xl bg-gray-200 disabled:opacity-50"
                disabled=""
              >
                <img
                  alt="Arrow Left"
                  class="w-5 h-5"
                  src=""
                />
              </button>
              <span
                class="px-3"
              >
                1 / 1
              </span>
              <button
                class="px-2 py-2 rounded-xl bg-gray-200 disabled:opacity-50"
                disabled=""
              >
                <img
                  alt="Arrow Right"
                  class="w-5 h-5"
                  src=""
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
